### PR TITLE
[bitnami/redmine] Chart standardized

### PR DESCRIPTION
**Pull request by juan131**
Signed-off-by: juan131 <juanariza@vmware.com>

**Description of the change**

This PR standardizes the `bitnami/redmine` chart to be in line with the rest of the catalog.

**Benefits**

Standardization.

**Possible drawbacks**

None. No major bump is required since breaking changes were added support for deprecated values using `coalesce`.

**Applicable issues**

N/A

**Additional information**

N/A

**Checklist**

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the values.yaml and added to the README.md using (readme-generator-for-helm)[https://github.com/bitnami-labs/readme-generator-for-helm]
- [x] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])

### CODE REVIEW

File: bitnami/redmine/templates/deployment.yaml [hunk 1]
```diff
@@ -174,46 +179,36 @@ spec:
             {{- end }}
           ports:
             - name: http
-              containerPort: {{ .Values.containerPort }}
+              containerPort: {{ coalesce .Values.containerPorts.http .Values.containerPort }}
+          {{- if not .Values.diagnosticMode.enabled }}
           {{- if .Values.livenessProbe.enabled }}
-          livenessProbe:
+          livenessProbe: {{- omit .Values.livenessProbe "enabled" "path" | toYaml | nindent 12 }}
```
Comment (migruiz4) [inline bitnami/redmine/templates/deployment.yaml:185]: Why are you no longer using?
```
{{- include "common.tplvalues.render" (dict "value" (omit .Values.livenessProbe "enabled" "path") "context" $) | nindent 12 }}
```

File: bitnami/redmine/templates/deployment.yaml [hunk 2]
```diff
@@ -233,7 +228,7 @@ spec:
         - name: redmine-data
           {{- if .Values.persistence.enabled }}
           persistentVolumeClaim:
-            claimName: {{ .Values.persistence.existingClaim | default (include "common.names.fullname" .) }}
+            claimName: {{ (tpl .Values.persistence.existingClaim $) | default (include "common.names.fullname" .) }}
```
Comment (migruiz4) [inline bitnami/redmine/templates/deployment.yaml:231]:
```suggestion
            claimName: {{ default (include "common.names.fullname" .) (tpl .Values.persistence.existingClaim $) }}
```

File: bitnami/redmine/templates/svc.yaml [hunk 3]
```diff
@@ -18,30 +19,33 @@ metadata:
   {{- end }}
 spec:
   type: {{ .Values.service.type }}
-  sessionAffinity: {{ default "None" .Values.service.sessionAffinity }}
+  sessionAffinity: {{ .Values.service.sessionAffinity }}
```
Comment (migruiz4) [inline bitnami/redmine/templates/svc.yaml:22]: If removed `default "None"`, shouldn't this be inside a conditional?

File: bitnami/redmine/templates/svc.yaml [hunk 4]
```diff
@@ -18,30 +19,33 @@ metadata:
...
   {{- end }}
   {{- if (or (eq .Values.service.type "LoadBalancer") (eq .Values.service.type "NodePort")) }}
   externalTrafficPolicy: {{ .Values.service.externalTrafficPolicy | quote }}
   {{- end }}
-  {{- if (and (eq .Values.service.type "LoadBalancer") .Values.service.loadBalancerSourceRanges) }}
-  loadBalancerSourceRanges: {{- toYaml . | nindent 4 }}
+  {{- if eq .Values.service.type "LoadBalancer" }}
+  {{- with .Values.service.loadBalancerSourceRanges }}
+  loadBalancerSourceRanges: {{ .Values.service.loadBalancerSourceRanges }}
   {{- end }}
```
Comment (migruiz4) [inline bitnami/redmine/templates/svc.yaml:32]: Why the change from
```
  {{- if and (eq .Values.service.type "LoadBalancer") .Values.service.loadBalancerSourceRanges }}
  loadBalancerSourceRanges: {{- toYaml .Values.service.loadBalancerSourceRanges | nindent 4 }}
  {{- end }}
```

File: bitnami/redmine/values.yaml [hunk 5]
```diff
@@ -396,18 +422,11 @@ ingress:
...
   ##
   enabled: false
-  ## @param ingress.className Set the ingressClassName which will be used for this ingress
-  ## The newer ingressClassName field on Ingresses is a replacement for that "kubernetes.io/ingress.class" annotation (Kubernetes 1.18+).
-  ## When using this filed you need to create the IngressClass resource with the corresponding name.
+  ## @param ingress.className IngressClass that will be be used to implement the Ingress (Kubernetes 1.18+)
+  ## This is supported in Kubernetes 1.18+ and required if you have more than one IngressClass marked as the default for your cluster .
+  ## ref: https://kubernetes.io/blog/2020/04/02/improvements-to-the-ingress-api-in-kubernetes-1.18/
   ##
   className: ""
```
Comment (migruiz4) [inline bitnami/redmine/values.yaml:429]: Could you please add support for `ingressClassName` using `coalesce`? Don't want to bump a major, but would like the charts to at least support the same values.

File: bitnami/redmine/values.yaml [hunk 6]
```diff
@@ -812,30 +757,25 @@ mailReceiver:
   ## @param mailReceiver.allowOverride Defines if email content is allowed to set attributes values. Values is a comma separated list of attributes or `all` to allow all attributes
   ##
   allowOverride: ""
+  ## @param mailReceiver.command Override default container command (useful when using custom images)
+  ##
+  command: ["bash"]
+  ## @param mailReceiver.args Override default container args (useful when using custom images)
+  ##
+  args: ["-c", "/cj/receive.sh"]
```
Comment (migruiz4) [inline bitnami/redmine/values.yaml:765]: If we are setting here `["bash"]` and `["-c", "/cj/receive.sh"]`,  then these values aren't really meant to override default as their description says.

I think we should move these inside the Deployment template, and leave them empty `[""]` for users who'd like to overwrite them.

**Review by migruiz4 — CHANGES_REQUESTED** [review]
Please check my comments below.

In addition, I would appreciate it if you could also have a look at the `cronjob.yaml`.

File: bitnami/redmine/templates/deployment.yaml (hunk 2 above)
Comment (juan131) [reply bitnami/redmine/templates/deployment.yaml:231]: It's exactly the same thing

File: bitnami/redmine/templates/svc.yaml (hunk 3 above)
Comment (juan131) [reply bitnami/redmine/templates/svc.yaml:22]: Nope. `service.sessionAffinity` is set by default in the values.yaml

**Review by migruiz4 — DISMISSED** [review]
